src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,175 @@
+    private User john;
+    private User jane;
+    private User maria;
+    private User george;
+    private String validPassword;
+    private String anotherValidPassword;
+    private String invalidPassword;
+    private String shortPassword;
+    private String validLogin;
+    private String anotherValidLogin;
+    private String thirdValidLogin;
+    private String invalidLogin;
+    private String shortLogin;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        john = new User();
+        jane = new User();
+        maria = new User();
+        george = new User();
+        validPassword = "qwerty";
+        anotherValidPassword = "qwe123rty";
+        invalidPassword = "abc12";
+        shortPassword = "ox";
+        validLogin = "UserName";
+        anotherValidLogin = "AcceptableLogin";
+        thirdValidLogin = "MyName";
+        invalidLogin = "NameX";
+        shortLogin = "Rw";
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_DifferentValidUsers_Ok() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        maria.setAge(ANOTHER_ADULT_AGE);
+        maria.setLogin(anotherValidLogin);
+        maria.setPassword(anotherValidPassword);
+        User actual = registrationService.register(john);
+        assertEquals(actual, john);
+        User nextUserToAdd = registrationService.register(maria);
+        assertEquals(nextUserToAdd, maria);
+    }
+
+    @Test
+    void register_underageUser_NotOk() {
+        john.setAge(TEEN_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_inadmissibleAge_NotOK() {
+        george.setAge(ZERO_AGE);
+        george.setLogin(anotherValidLogin);
+        george.setPassword(anotherValidPassword);
+        jane.setAge(NEGATIVE_AGE);
+        jane.setLogin(thirdValidLogin);
+        jane.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(jane));
+    }
+
+    @Test
+    void register_shortLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(invalidLogin);
+        john.setPassword(validPassword);
+        george.setAge(ANOTHER_ADULT_AGE);
+        george.setLogin(shortLogin);
+        george.setPassword(anotherValidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_shortPassword_NotOk() {
```

Comment:
Similar to the previous comment, make sure to include edge case tests for password length. For example, test a password that is exactly 6 characters long to ensure that the boundary condition is correctly handled.